lib/pos_mempool.go

```diff
@@ -164,20 +165,16 @@ func (dmp *PosMempool) Stop() {
 	dmp.txnRegister.Reset()
 	dmp.ledger.Reset()
 
-	dmp.status = PosMempoolStatusNotRunning
+	dmp.status.Store(PosMempoolStatusNotRunning)
 }
 
 func (dmp *PosMempool) IsRunning() bool {
```

Comment:
Does the `PosMempool.IsRunning()` function have any expected use-cases outside of this class? Similar to the Fast-HotStuff consensus module, it seems like simply exposing the `Start()` and `Stop()` results in the simplest public interface

Reply:
Nah doesn't really have any expected public use-cases. Maybe it should be private. It's cool that you can use it instead of manual status checks and for tests.

Reply:
I'd suggest removing it then. It's valid for the mempool struct's unit tests to have full knowledge of the mempool's inner workings, i.e. to be able to inspect statuses manually.